util/types/json/impl.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+func parseFromString(s string) (JSON, error) {
+	if len(s) == 0 {
+		return nil, ErrInvalidJSONText.GenByArgs("The document is empty")
+	}
+	var in interface{}
+	if err := json.Unmarshal([]byte(s), &in); err != nil {
+		return nil, ErrInvalidJSONText.GenByArgs(err)
+	}
+	return normalize(in), nil
+}
+
+func dumpToString(j JSON) string {
+	bytes, _ := json.Marshal(j)
+	return strings.Trim(string(bytes), "\n")
+}
+
+func normalize(in interface{}) JSON {
```

Comment:
we explicit implement `JSON` for `map[string]JSON` but not `map[string]interface{}`, so we need convert interface{} to JSON.